### ISSUE TITLE
Feature Request: Custom Field DueDate

### ISSUE THREAD

**Issue by xet7** [wekan/wekan#5621]
Moved to here from https://github.com/wekan/wekan/discussions/5608

From @mimZD

![image](https://github.com/user-attachments/assets/54cef9c8-dceb-4586-9567-ae95c3ef792d)

a) It would be great if we could choose other options instead of the "current date/time", a dropdown list with these options:

1. current date/time
2. tomorrow ( 24 hours later )
3. next week ( 7 days later )\
4. custom day: ==> in this option a text box with another dropdown list will appear
    [ empty text box ] [dropdown ( minute/hour/day/week/month) ]

option 4 is in order to change duedate into a custom duration

b) and triggers a rule by date ==>ex:  if a card is overdue then ...

**Comment by mimZD** [comment]
@xet7 
its not custom filed, its rule

**Comment by mimZD** [comment]
it could be like this:
![image](https://github.com/user-attachments/assets/59f0d87a-b941-4856-8118-a2653aebc694)


and customTime:

![image](https://github.com/user-attachments/assets/2c083d38-683d-4316-8499-e8bcb31b7fc4)

**Comment by mimZD** [comment]
@xet7 
and for duedate trigger:

![image](https://github.com/user-attachments/assets/a90e52b4-c100-417e-862e-e8dcd26a56aa)

**Comment by mimZD** [comment]
@xet7 
can i use webhook as a temp way to do this?
is there an doc or example for how to use webhook in wekan?

**Comment by xet7** [comment]
@mimZD 

If card is moved, etc, it's possible to send that event to webhook, at Board Settings / Webhooks, add URL to some webhook receiver.

For example, URL to this kind of webhook receiver, that uses WeKan with API:

https://github.com/wekan/webhook/blob/main/public/index.php#L46

https://github.com/wekan/wekan/blob/main/api.py

For scheduled tasks, run cron scheduled job, for example, bash script, that runs that WeKan API script. For example, edit cron task like this, and have that script executeable like this: `chmod +x something.sh`

List cron jobs:
```
crontab -l
```
Edit cron job:
```
sudo apt-get -y install nano
export EDITOR=nano
crontab -e
```
This script runs every 5 minutes:
```
# m h  dom mon dow   command
*/5 * * * * /home/wekan/something.sh > /dev/null 2>&1
```
Save and Exit with: Ctrl-o Enter Ctrl-x

Examples of cron scheduling: https://crontab.guru

**Comment by xet7** [comment]
More info: https://github.com/wekan/wekan/tree/main/docs/Webhooks

Some more Python based features: https://github.com/wekan/wekan/blob/main/docs/Features/Python.md

**Comment by mimZD** [comment]
@xet7 
thanks a lot 🥰🥰🥰🥰🥰🥰🥰🥰


only this remaining:



> [@xet7](https://github.com/xet7) it could be like this: ![image](https://github.com/user-attachments/assets/59f0d87a-b941-4856-8118-a2653aebc694)
> 
> and customTime:
> 
> ![image](https://github.com/user-attachments/assets/2c083d38-683d-4316-8499-e8bcb31b7fc4)

**Comment by mimZD** [comment]
@xet7 
no wait

you have done both trigger and action?

**Comment by xet7** [comment]
@mimZD 

Yes, there is setDateRelative action and the scheduleKind: 'due' (overdue/soon/set) trigger.
